packages/analytics-ma-browser/test/e2e.test.ts

```diff
@@ -0,0 +1,318 @@
+  });
+
+  test('should track attribution', () => {
+    setCookie('AMP_MKTG_API_KEY', {
+      utm_campaign: '123',
+    });
+
+    let requestBody: Record<string, any> = {};
+    const scope = nock(url)
+      .post(path, (body: Record<string, any>) => {
+        requestBody = body;
+        return true;
+      })
+      .reply(200, success);
+
+    const client = createInstance();
+    client.init('API_KEY', undefined);
+
+    return new Promise<void>((resolve) => {
+      scope.on('replied', () => {
+        expect(requestBody).toEqual({
+          api_key: 'API_KEY',
+          events: [
+            {
+              device_id: expectString,
+              event_id: 0,
+              event_type: '$identify',
+              insert_id: expectString,
+              ip: '$remote',
+              language: 'en-US',
+              library: expectLibrary,
+              os_name: 'WebKit',
+              os_version: '537.36',
+              platform: 'Web',
+              session_id: expectNumber,
+              time: expectNumber,
+              user_properties: {
+                $setOnce: {
+                  initial_fbclid: 'EMPTY',
+                  initial_gclid: 'EMPTY',
+                  initial_referrer: 'EMPTY',
+                  initial_referring_domain: 'EMPTY',
+                  initial_utm_campaign: 'EMPTY',
+                  initial_utm_content: 'EMPTY',
+                  initial_utm_medium: 'EMPTY',
+                  initial_utm_source: 'EMPTY',
+                  initial_utm_term: 'EMPTY',
+                },
+                $unset: {
+                  fbclid: '-',
+                  gclid: '-',
+                  referrer: '-',
+                  referring_domain: '-',
+                  utm_campaign: '-',
+                  utm_content: '-',
+                  utm_medium: '-',
+                  utm_source: '-',
+                  utm_term: '-',
+                },
+              },
+            },
+          ],
+          options: {},
+        });
+        scope.done();
+        resolve();
+      });
+    });
+  });
+
+  test('should track page view on attribution', () => {
+    setCookie('AMP_MKTG_API_KEY', {
+      utm_campaign: '123',
+    });
+
+    let requestBody: Record<string, any> = {};
+    const scope = nock(url)
+      .post(path, (body: Record<string, any>) => {
+        requestBody = body;
+        return true;
+      })
+      .reply(200, success);
+
+    const client = createInstance();
+    client.init('API_KEY', undefined, {
+      trackPageViews: {
+        trackOn: 'attribution',
+      },
+    });
+
+    return new Promise<void>((resolve) => {
+      scope.on('replied', () => {
+        expect(requestBody).toEqual({
+          api_key: 'API_KEY',
+          events: [
+            {
+              device_id: expectString,
+              event_id: 0,
+              event_properties: {
+                page_location: 'http://localhost/',
+                page_path: '/',
+                page_title: '',
```

Comment:
would expect `page_domain` and `page_url` to also appear here
![image](https://user-images.githubusercontent.com/9955575/192063894-096cda12-13d6-45e6-b563-07fe8fb1b555.png)

Reply:
correct it should be there. let me check. might be because it has not been rebased yet

Reply:
i updated the test and these field are being asserted including the additional click ids